include/solvers/navier_stokes_scratch_data.h

```diff
@@ -769,64 +783,132 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
+  /** @brief Calculates the properties of the fluid at the locations of the particles.
+   * At the moment, only constant properties within the same fluid are
+   * supported. When two fluids are present and VOF is used, the properties are
+   * calculated based on the filtered phase fraction interpolated at the
+   * location of the particles. These properties are used in the forces
+   * calculations in the VANS equations.
+   */
 
-        fe_values_particles_void_fraction.get_function_values(
-          void_fraction_solution, cell_void_fraction);
+  void
+  calculate_fluid_properties_at_particle_location()
+  {
+    density_at_particle_location.resize(number_of_particles);
+    kinematic_viscosity_at_particle_location.resize(number_of_particles);
+
+    if (gather_vof)
+      {
+        for (unsigned int particle_i = 0; particle_i < number_of_particles;
+             ++particle_i)
+          {
+            density_at_particle_location[particle_i] = calculate_point_property(
+              filtered_phase_values_at_particle_location[particle_i],
+              this->density_ref_0,
+              this->density_ref_1);
+            double dynamic_viscosity_at_particle_location =
+              calculate_point_property(
+                filtered_phase_values_at_particle_location[particle_i],
+                this->kinematic_viscosity_scale_0 * this->density_ref_0,
+                this->kinematic_viscosity_scale_1 * this->density_ref_1);
+
+            kinematic_viscosity_at_particle_location[particle_i] =
+              dynamic_viscosity_at_particle_location /
+              density_at_particle_location[particle_i];
+          }
       }
+    else
+      {
+        for (unsigned int particle_i = 0; particle_i < number_of_particles;
+             ++particle_i)
+          {
+            // Calculate the kinematic viscosity and density at the particle
+            // location assuming a constant kinematic viscosity and density in a
+            // single fluid
+            kinematic_viscosity_at_particle_location[particle_i] =
+              kinematic_viscosity_scale;
+            density_at_particle_location[particle_i] = density_scale;
+          }
+      }
+  }
 
-    // Relative velocity and particle Reynolds
+  /** @brief Calculates the velocity of the fluid relative to that of the particle and
+   * the particle Reynolds number at the location of the particles. These will
+   * be used in forces calculations in the vans equations.
+   */
+
+  void
+  calculate_force_parameters_at_particle_location()
+
+  { // Relative velocity and particle Reynolds
     unsigned int particle_i                  = 0;
     average_fluid_particle_relative_velocity = 0;
 
-    // TODO, get the real viscosity at the particle localtion
-    double kinematic_viscosity =
-      properties_manager.get_kinematic_viscosity_scale();
-
+    // At the moment, the kinematic viscosity and density are taken to be
```

Comment:
Maybe I would add this in the "brief". What do you think?

Reply:
Agreed. Good point.